jenkins/validate-artifacts/validate-artifacts.jenkinsfile

```diff
@@ -118,7 +138,21 @@ pipeline {
 
                     if(OPENSEARCH_ARTIFACT_URL) {
                         ARCHITECTURE = url_paths.values().find { it.contains("x64") } ? "x64" : "arm64"
-                        DISTRIBUTION = url_paths.values().find { it.contains("tar.gz") } ? "tar" : (url_paths.values().find { it.contains("repo") } ? "yum" : "rpm")
+                        def distKeyWords = [
+                            "tar.gz": "tar",
+                            "zip": "zip",
+                            "rpm": "rpm",
+                            "repo": "yum",
```

Comment:
Why is repo: yum? How would you handle apt then?

Reply:
Currently we do not support apt. I can refactor code once we add apt changes. And this mapping is accessed only  when user provides only artifact url links to validate.

Reply:
And that's a keyword to execute "yum" child class when yum artifacts are provided, for apt we can use "apt" As keyword

Reply:
Why cant we use yum to replace `repo`?

Reply:
In offline discussion this is to get a keyword for finding the distro through url:
`https://artifacts.opensearch.org/releases/bundle/opensearch/2.x/opensearch-2.x.repo`.

Think there can be a better solution for improvement later.